packages/effect/src/internal/pubsub.ts

```diff
@@ -137,6 +138,12 @@ export const publishAll = dual<
 export const subscribe = <A>(self: PubSub.PubSub<A>): Effect.Effect<Queue.Dequeue<A>, never, Scope.Scope> =>
   self.subscribe
 
+/** @internal */
+export const withReplay = dual<
+  (n: number) => <A>(self: PubSub.PubSub<A>) => PubSub.PubSub<A>,
```

Comment:
What do you think about 
```ts
  (n: number) => <A>(self: Effect.Effect<PubSub.PubSub<A>>) => Effect.Effect<PubSub.PubSub<A>>,
```

So we can write

```ts
const replayPubSub = yield* withReplay(3)(PubSub.unbounded())
```

Because 
```ts
const pubsub = yield* PubSub.unbounded<number>().pipe(
    Effect.map(PubSub.withReplay(3))
  )
```
looks verbose to me

Reply:
Went with a constructor option.